portal-impl/src/com/liferay/portal/service/impl/ContactLocalServiceImpl.java

```diff
@@ -211,7 +211,8 @@ protected void validate(String className, long classPK)
 	protected void validateBirthday(Date birthday) {
 		if ((birthday != null) && birthday.after(new Date())) {
 			throw new SystemException(
-				"Birthday cannot be set to a date in future.");
+				new ContactBirthdayException(
```

Comment:
This is wrong. You should not encapsulate ContactBirthdayExcecption inside a SystemException, since it is a real expected exception.

I've checked that all methods but one (addContact) are already throwing PortalException so this should be doable without API changes.

In that one method, we can do the wrapping as a last resource, although that method should not be invoked at all anyway.